warehouse/query-core/src/main/java/datawave/query/jexl/lookups/LookupTermsFromRegex.java

```diff
@@ -256,15 +258,25 @@ public IndexLookupMap lookup(ShardQueryConfiguration config, ScannerFactory scan
                             topKey.getColumnFamily(holder);
                             String field = holder.toString();
                             
-                            // We are only returning a mapping of field value to field name, no need to
-                            // determine cardinality and such at this point.
-                            fieldsToValues.put(field, term);
-                            // conditional states that if we exceed the key threshold OR field name is not null and we've exceeded
-                            // the value threshold for that field name ( in the case where we have a fielded lookup ).
-                            if (fieldsToValues.isKeyThresholdExceeded() || (fields.size() == 1 && fieldsToValues.get(field).isThresholdExceeded())) {
-                                if (log.isTraceEnabled())
-                                    log.trace("We've passed term expansion threshold");
-                                return true;
+                            // if the limits haven't already been surpassed
+                            if (!fieldsToValues.isKeyThresholdExceeded()
+                                            && (fieldsToValues.get(field) == null || !fieldsToValues.get(field).isThresholdExceeded())) {
+                                // We are only returning a mapping of field value to field name, no need to
+                                // determine cardinality and such at this point.
+                                boolean added = fieldsToValues.put(field, term);
+                                
+                                // if there was a failure to add, and the key threshold isn't exceeded, increment the value exceeded count
+                                if (!added && !fieldsToValues.isKeyThresholdExceeded()) {
+                                    valueExceededCount++;
```

Comment:
I don't think this is correct.  Using the return value from fieldsToValues.put being false does not mean that fieldsToValues.get(field).isThresholdExceeded().  The return value could be false because that value was already in the set.

Reply:
you are right, if the key is already present in the underlying value hashset it will return false.